core/test/Assert.re

```diff
@@ -123,13 +123,20 @@ let update =
     ),
 };
 
-let flushPendingUpdates = ({renderedElement, syntheticElement}) => {
-  syntheticElement,
-  renderedElement: RenderedElement.flushPendingUpdates(renderedElement),
-};
+let flushPendingUpdates = ({renderedElement, syntheticElement} as testState) =>
+  Implementation.isDirty^ ?
+    {
+      Implementation.isDirty := false;
+      {
+        syntheticElement,
+        renderedElement: RenderedElement.flushPendingUpdates(renderedElement),
+      };
+    } :
+    testState;
 
 let executeSideEffects = ({renderedElement} as testState) => {
   RenderedElement.executeHostViewUpdates(renderedElement) |> ignore;
+  List.iter(f => f(), renderedElement.enqueuedEffects);
```

Comment:
It looks like `enqueuedEffects` isn't available on the `renderedElement` via `ReactCore`.

Is it expected that these effects would be ran as part of `executeHostViewUpdates`? Or will there be a separate API, like `executeEffects`?

Reply:
There has to be a separate API. The reason is that we normally want to call `executeHostViewUpdates` on the main thread but the effects should be executed on a background one.